src/main/haskell/language-kore/app/parser/Main.hs

```diff
@@ -2,102 +2,135 @@ module Main where
+import           Data.Semigroup                           ((<>))
+
+{- 
+Main module to run kore-parser
+TODO: add command line argument tab-completion
+-}
+
+-- | Main options record    
+data KoreParserOptions = KoreParserOptions
+    { fileNameArgument  :: String -- ^ Filename to parse and verify 
+    , willPrintOption   :: Bool   -- ^ Option to print definition
+    , willVerifyOption  :: Bool   -- ^ Option to verify definition
+    , willChkAttrOption :: Bool   -- ^ Option to check attributes during verification
     }
 
-usage :: String
-usage =
-    "Usage: kore-parser [--[no]verify] [--[no]print] fileName"
-
--- TODO(virgil): Use a generic command line parsing library instead.
-parseCommandLineFlags :: [String] -> CommandLineFlags
-parseCommandLineFlags [] =
-    CommandLineFlags
-        { commandLineFlagsFileName = Nothing
-        , commandLineFlagsVerify = True
-        , commandLineFlagsPrint = True
-        }
-parseCommandLineFlags (('-' : '-' : firstFlag) : commandLineReminder) =
-    addFlagAndContinue firstFlag commandLineReminder
+                    
+{-| Command Line Argument Parser -}
+commandLineParser :: Parser KoreParserOptions
+commandLineParser = KoreParserOptions
+      <$> argument str
+              ( metavar "FILE"
+              <> help "Kore source file to parse [and verify]" )
+      <*> switch3 "print"
+              "Print parsed definition to stdout" 
+              "Do not print parsed definition to stdout" 
+      <*> switch3 "verify"
+              "Verify well-formedness of parsed definition"
+              "Do not verify well-formedness of parsed definition" 
+      <*> switch3 "chkattr"
+              "Check attributes during verification"
+              "Ignore attributes during verification"
+
+
+-- | Run argument parser for kore-parser
+commandLineParse :: IO KoreParserOptions
+commandLineParse = execParser opts
   where
-    addFlagAndContinue flag commandLine
-        | flag == "verify" =
-            (parseCommandLineFlags commandLine)
-                { commandLineFlagsVerify = True }
-        | flag == "noverify" =
-            (parseCommandLineFlags commandLine)
-                { commandLineFlagsVerify = False }
-        | flag == "print" =
-            (parseCommandLineFlags commandLine)
-                { commandLineFlagsPrint = True }
-        | flag == "noprint" =
-            (parseCommandLineFlags commandLine)
-                { commandLineFlagsPrint = False }
-        | otherwise =
-            error ("Unknown flag: --" ++ flag)
-parseCommandLineFlags (flag : commandLine) =
-    (parseCommandLineFlags commandLine) { commandLineFlagsFileName = Just flag }
+    opts = info (commandLineParser <**> helper)
+      ( fullDesc
+     <> progDesc "Parses Kore definition in FILE; optionally, \
+                 \Verifies well-formedness"
+     <> header "kore-parser - a parser for Kore definitions" )
+
+
+main :: IO ()
+main =
+    do {
+    ; KoreParserOptions
```

Comment:
I have mixed feelings about this way of writing 'do' statements because it is fairly different from what we currently use, and consistency is nice. On the other hand, it looks nicer, because it's easier to see what is a top-level statement in a 'do'. @traiansf , what do you think?

Reply:
Maybe only in complicated 'do' statements? We don't need so much syntax when it's a straightforward 'do' block.  It just seems useful when we have to mix other nested language constructs. Like in 'main' definitions.

Reply:
I allowed this usage in a restricted way (i.e. when it actually helps), see the style guide doc, Identation, Do. If you think that's not reasonable, please say so (in which case an example showing why the style guide is not reasonable would help).